corva/middleware/unpack_context.py

```diff
@@ -0,0 +1,24 @@
+from typing import Callable
+
+from corva.models.base import BaseContext
+
+
+def unpack_context_factory(include_state=False):
+    def unpack_context(context: BaseContext, call_next: Callable) -> BaseContext:
```

Comment:
can't we just move the `include_state` in the `unpack_context` function as a parameter and get rid of the nested function?

Reply:
we can, then usage will look like
```python3
unpack_context_factory(include_state=True)  # before

functools.partial(unpack_context, include_state=True)  # after
```

Reply:
Why do we need to have a function in the result?. Why not return value right away?

Reply:
we need the function to wrap it into middleware layers. after wrapping middleware stack will be called and run our function at some point

Reply:
Can we pass either a context or a standard interface middleware functions, and expect them to all implement the interface, instead of having to do magic DI here?

Alternately, middleware could be classes instead of functions, and you pass in all these keywords to the __init__ call, and the middleware class can then choose whether it wants to deal with them or not.

Personally I think using middleware classes is easier to understand and use, and gives more powerful functionality.